pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -928,6 +943,27 @@ public void redeliverUnacknowledgedMessages(List<MessageIdImpl> messageIds) {
         }
     }
 
+    private int removeExpiredMessagesFromQueue(Set<MessageIdImpl> messageIds) {
+        int messagesFromQueue = 0;
+        for (Message message : incomingMessages) {
+            MessageIdImpl messageId = (MessageIdImpl) message.getMessageId();
+            if (messageId instanceof BatchMessageIdImpl) {
+                // messageIds contain MessageIdImpl, not BatchMessageIdImpl
+                messageId = new MessageIdImpl(messageId.getLedgerId(), messageId.getEntryId(),
+                        getPartitionIndex());
+            }
+            if (messageIds.contains(messageId)) {
```

Comment:
Yeah, that's true.. same comment as above, let's just clear the queue and ask to resend all messages